File: YSW-Yang/02/[PGS - Lv2] n^2 배열 자르기.java
class Solution {
    public int[] solution(int n, long left, long right) {
        int N = (int) (right - left + 1);
        int[] answer = new int[N];
        
        for(int i = 0; i < N; i++){
            int row = (int) ((left + i) / n);
            int col = (int) ((left + i) % n);
            
            answer[i] = Math.max(row, col) + 1;
        }
        
        return answer;
    }
}

File: 7월 3주차/JeongEon8/03/[PGS - Lv2] n^2 배열 자르기.cpp
#include <vector>

using namespace std;

vector<int> solution(int n, long long left, long long right) {
    vector<int> answer;

    for(long long i = left; i <= right; ++i) {
        long long x = i / n;
        long long y = i % n;
        if(x < y) answer.push_back(y + 1);
        else answer.push_back(x + 1);
    }
    return answer;
}
